crates/c_ext/src/sparse_observable.rs

```diff
@@ -11,13 +11,290 @@
 // that they have been altered from the originals.
 
 use num_complex::Complex64;
-use qiskit_accelerate::sparse_observable::SparseObservable;
+use qiskit_accelerate::sparse_observable::{BitTerm, SparseObservable, SparseTerm};
+
+/// A vector of indices as ``uint_32t``.
+#[derive(Debug, Clone)]
+pub struct IndexVec {
+    // We use a struct instead of a typedef (type IndexVec = Vec<u32>) to avoid
+    // cbindgen to introduce a redundant Vec_u32 type. This could maybe be changed in
+    // future.
+    values: Vec<u32>,
+}
+
+/// A vector of ``BitTerm`` enum values.
+#[derive(Debug, Clone)]
+pub struct BitTermVec {
+    // We use a struct instead of a typedef (type IndexVec = Vec<u32>) to avoid
+    // cbindgen to introduce a redundant Vec_u32 type. This could maybe be changed in
+    // future.
+    values: Vec<BitTerm>,
+}
+
+/// A Pauli term vector, containing ``(index, bit_term)`` tuples.
+#[derive(Debug, Clone)]
+pub struct PauliTermVec {
+    // We store the data as two separate vectors, which simplified constructing
+    // SparseTerms later on, which expect this format.
+    indices: Vec<u32>,
+    bit_terms: Vec<BitTerm>,
```

Comment:
```suggestion
    indices: IndexVec,
    bit_terms: BitTermVec,
```
Or do you see a reason not to do this?

Reply:
`IndexVec & BitTermVec` are removed now